jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/StepsRunner.java

```diff
@@ -151,7 +220,7 @@ private void pullAndCacheBaseImageLayers() {
         executorService.submit(
             () ->
                 scheduleCallables(
```

Comment:
I know this is old code, but why are submitting something that then submits some more things? Is this some kind of delayed submit? Could we just call scheduleCallables without the wrapper Callable?

Reply:
This scheduling depends on getting `results.baseImageAndAuth` (a `Future`), so calling `.get()` on it will block. Retrieving the base image should not be on the main thread of `StepsRunner`.